### PR TITLE
faster build

### PR DESCRIPTION
**Pull request by kof**
- properly restarts the electron app on changes
- builds and copies everything by just running npm run start

### CODE REVIEW

File: app/src/app/logToFile.js [hunk 1]
```diff
@@ -13,12 +13,22 @@ function escapePath(path) {
   return path.replace(/( |\(|\))/g, `${escapeChar}$1`)
 }
 
+function hookStream(stream1, callback) {
```
Comment (giuseppeg) [inline app/src/app/logToFile.js]: any reason why you called it `stream1` instead of `stream`?

File: tasks/start.js [hunk 2]
```diff
@@ -22,25 +26,43 @@ var runBuild = function () {
     });
 
     build.on('close', function (code) {
```
Comment (giuseppeg) [inline tasks/start.js]: nitpick: this could be `build.on('close', deferred.resolve)`

File: tasks/start.js [hunk 3]
```diff
@@ -4,13 +4,17 @@ var Q = require('q');
...
+var gutil = require('gulp-util');
 
 var gulpPath = pathUtil.resolve('./node_modules/.bin/gulp');
+var srcDir = jetpack.cwd('./app');
+var destDir = jetpack.cwd('./build');
+var watchDir = './lib/**'
 
-var runBuild = function () {
+function runBuildApp() {
     var deferred = Q.defer();
```
Comment (giuseppeg) [inline tasks/start.js]: why are we not using `Promise`?

File: tasks/start.js (hunk 2 above)
Comment (kof) [inline tasks/start.js]: is deferred.resolve already bound?

**Comment by giuseppeg** [comment]
@kof done, I have not much knowledge of Electron but the changes make sense.

File: tasks/start.js (hunk 3 above)
Comment (kof) [inline tasks/start.js]: has been used before I started to work on it, would change them all at once.

**Comment by kof** [comment]
merged

File: tasks/start.js (hunk 2 above)
Comment (giuseppeg) [inline tasks/start.js]: it doesn't have to be, it'll call any function with the same arguments that you get in the anonymous function

**Pull request merged**